cfme/tests/services/test_catalog.py

```diff
@@ -6,46 +6,50 @@
 from cfme.utils import error
 from cfme.utils.blockers import BZ
 from cfme import test_requirements
-from cfme.services.catalogs.catalog import Catalog
 from cfme.utils.update import update
 
 pytestmark = [test_requirements.service, pytest.mark.tier(2)]
 
 
 @pytest.mark.sauce
-def test_catalog_crud():
-    cat = Catalog(name=fauxfactory.gen_alphanumeric(),
-                  description="my catalog")
-    cat.create()
+def test_catalog_crud(appliance):
+    cat = appliance.collections.catalogs.create(name=fauxfactory.gen_alphanumeric(),
+                                                description="my catalog")
     with update(cat):
         cat.description = "my edited description"
     cat.delete()
 
 
 @pytest.mark.sauce
-def test_catalog_duplicate_name():
-    cat = Catalog(name=fauxfactory.gen_alphanumeric(),
-                  description="my catalog")
-    cat.create()
+def test_catalog_duplicate_name(appliance):
+    catalog_name = fauxfactory.gen_alphanumeric()
+    cat = appliance.collections.catalogs.create(name=catalog_name, description="my catalog")
     with error.expected("Name has already been taken"):
-        cat.create()
+        appliance.collections.catalogs.create(name=catalog_name, description="my catalog")
     cat.delete()
 
 
 @pytest.mark.meta(blockers=[BZ(1460891, forced_streams=['5.8', '5.9', 'upstream'])])
 @pytest.mark.sauce
 def test_permissions_catalog_add(appliance):
     """ Tests that a catalog can be added only with the right permissions"""
-    cat = Catalog(name=fauxfactory.gen_alphanumeric(),
-                  description="my catalog")
+    catalogs = []
+
+    def _create_catalog():
+        catalogs.append(appliance.collections.catalogs.create(name=fauxfactory.gen_alphanumeric(),
+                                                              description="my catalog"))
+
+    def _delete_catalog():
+        if catalogs:
+            catalogs[0].delete()
```

Comment:
Why only delete the first one here, and not iterate over catalogs? (If there's a good reason put in-line in comment please :) )